packages/embark-deploy-tracker/src/trackingFunctions.js

```diff
@@ -0,0 +1,110 @@
+import {__} from 'embark-i18n';
+import {dappPath, sha3} from 'embark-utils';
+import Web3 from 'web3';
+
+export default class TrackingFunctions {
+  constructor({config, env, fs, events, logger, trackContracts}) {
+    this.config = config;
+    this.chainConfig = {};
+    this.chainFile = config.contractsConfig.tracking;
+    this.currentChain = null;
+    this.env = env;
+    this.fs = fs;
+    this.events = events;
+    this.logger = logger;
+    this._web3 = null;
+    this.trackContracts = (trackContracts !== false);
+  }
+
+  get web3() {
+    return (async () => {
+      if (!this._web3) {
+        const provider = await this.events.request2("blockchain:client:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  getContract(name, code, args) {
+    if (!this.currentChain) return false;
+    let contract = this.currentChain.contracts[sha3(code + name + args.join(','))];
+    if (contract && contract.address === undefined) {
+      return false;
+    }
+    return contract;
+  }
+
+  trackAndSaveContract(params, cb) {
+    const {contract} = params;
+    if (!this.chainFile || !this.trackContracts || contract.track === false) return cb();
+    this.trackContract(contract.className, contract.realRuntimeBytecode, contract.realArgs || contract.args, contract.deployedAddress);
+    this.save();
+    cb();
+  }
+
+  loadChainTrackerFile() {
+    if (this.chainFile === false) return;
+    if (this.chainFile === undefined) this.chainFile = ".embark/chains.json";
+    this.chainFile = dappPath(this.chainFile);
+    if (!this.fs.existsSync(this.chainFile)) {
+      this.logger.info(this.chainFile + ' ' + __('file not found, creating it...'));
+      this.fs.outputJSONSync(this.chainFile, {});
+    }
+
+    this.chainConfig = this.fs.readJSONSync(this.chainFile);
+  }
+
+  setCurrentChain(_params, callback) {
+    if (!this.chainFile) return callback();
+    if (this.chainFile === false) return callback();
```

Comment:
The second line is useless. If chainFile is `false`, it will trigger the first `if`

Reply:
Removed the first line, good catch!